online-shopping-website/frontend/src/Components/Products/ProductPage.js

```diff
@@ -3,6 +3,7 @@ import Grid from '@mui/material/Grid';
 import { ProductGrid } from './ProductGrid';
 import { SideNav } from './SideNav';
 import axios from 'axios';
+import * as React from "react";
```

Comment:
Please only import what's needed from react, instead of importing everything which is heavy.
For example, if you need `useEffect` and `useState` hooks, just do `import { useEffect, useState } from 'react';` to avoid unnecessary large imports.

Reply:
Totally agree, must've missed this one. Nice catch!